### PR TITLE
UHM-5284 Update HIV Intake prophylaxis treatment htmlform

### PR DESCRIPTION
**Pull request by brandones**
https://pihemr.atlassian.net/browse/UHM-5284

Screenshot from 2021-03-12 10-55-34

### CODE REVIEW

**Comment by brandones** [comment]
@lnball The horizontal scrollbar is not awesome, but I think our options are that or getting rid of the table format altogether and just making each Prophylaxis display like a normal grouping of questions.

**Review by lnball — COMMENTED** [review]
Why aren't we using Drug orders for prophylaxis?  Is that what we want for treatments?  We have all the question there -- except primary/secondary for Isoniazid.  That could be a radio button and separate question added to this section.

If not, any chance you can merge columns (ie. start|continue|stop and use radio buttons/newline for these 3?) so you don't end up with the scrollbar.  'Start' toggles the start date.  'Stop' toggles the reason.
- start
- stop
- continue

@brandones @mogoodrich

**Comment by brandones** [comment]
@mseaton or @mogoodrich , can you comment on whether we should be using the Drug orders tag for this? The information we need to collect is this:

image

**Comment by mogoodrich** [comment]
It's a good question about using drug orders going forward... however, regardless, we will want the v3 form to be observation-based so that we can migrate in the legacy  data properly.

Once we get this looking the way we it, let's merge this in, keeping both the v3 and v4 forms obs-based (I think you just copied the identifical code between the tw foro, correct @brandones ) and then discuss/create a new ticket for switching v4 to drug order tags.

**Comment by mseaton** [comment]
Agreed @mogoodrich .  Let's definitely create a ticket for converting the v4 form to drug orders.  That definitely makes the most sense there I think.

**Comment by brandones** [comment]
Screenshot from 2021-03-12 10-55-34

File: configuration/pih/htmlforms/hiv/section-hiv-plan.xml [hunk 1]
```diff
@@ -498,161 +502,200 @@
...
+                                                toggle="{id: '{drugname}-treat', style: 'dim'}"/>
+                                        </td>
+                                        <td class="{drugname}-treat">
+                                            <obs conceptId="PIH:PROPHYLAXIS STARTED"
+                                                 answerConceptId="CIEL:1065" style="checkbox"
+                                                 labelCode="pihcore.hiv.started" answerLabel=""
+                                                 toggle="{drugname}-start-date"/>
+                                            <obs conceptId="PIH:Prophylaxis should continue"
+                                                 answerConceptId="CIEL:1065" style="checkbox"
+                                                 labelCode="pihcore.hiv.continue" answerLabel="" />
```
Comment (lnball) [inline configuration/pih/htmlforms/hiv/section-hiv-plan.xml:544]: Use labelCode="emr.continue"

File: configuration/pih/htmlforms/hiv/section-hiv-plan.xml [hunk 2]
```diff
@@ -498,161 +502,200 @@
...
+                                        <td>
+                                            <!-- Medication orders -->
+                                            <obs conceptId="CIEL:1282" answerConceptId="{conceptMap}"
+                                                answerLabel="{drugname}"
+                                                toggle="{id: '{drugname}-treat', style: 'dim'}"/>
+                                        </td>
+                                        <td class="{drugname}-treat">
+                                            <obs conceptId="PIH:PROPHYLAXIS STARTED"
+                                                 answerConceptId="CIEL:1065" style="checkbox"
+                                                 labelCode="pihcore.hiv.started" answerLabel=""
```
Comment (lnball) [inline configuration/pih/htmlforms/hiv/section-hiv-plan.xml:540]: Use labelCode="general.start"

File: configuration/pih/htmlforms/hiv/section-hiv-plan.xml [hunk 3]
```diff
@@ -498,161 +502,200 @@
...
+                                                 toggle="{drugname}-start-date"/>
+                                            <obs conceptId="PIH:Prophylaxis should continue"
+                                                 answerConceptId="CIEL:1065" style="checkbox"
+                                                 labelCode="pihcore.hiv.continue" answerLabel="" />
+                                            <obs conceptId="PIH:Stop Rx" style="checkbox"
+                                                 answerConceptId="CIEL:1065"
+                                                 labelCode="pihcore.hiv.stop" answerLabel=""
```
Comment (lnball) [inline configuration/pih/htmlforms/hiv/section-hiv-plan.xml:547]: Cannot find a handy translation.  Change labelCode="pihcore.stop".  There's nothing about this specific to HIV.

File: configuration/pih/htmlforms/hiv/section-hiv-plan.xml [hunk 4]
```diff
@@ -498,161 +502,200 @@
...
+                                <label>
+                                    <uimessage code="pihcore.hiv.statusAndOrder" />
+                                </label>
                             </td>
-                            <td class="other-pro-treat">
-                                <obs conceptId="CIEL:159367" answerConceptId="CIEL:1065"
-                                     answerLabel=""/>
+                            <td>
+                                <label>
+                                    <uimessage code="pihcore.hiv.dosage" />
```
Comment (lnball) [inline configuration/pih/htmlforms/hiv/section-hiv-plan.xml:520]: Same here.  Don't include hiv in the message code.

File: configuration/pih/htmlforms/hiv/section-hiv-plan.xml [hunk 5]
```diff
@@ -498,161 +502,200 @@
...
-                                <obs conceptId="CIEL:163526" />
+                                <label>
+                                    <uimessage code="pihcore.treatment" />
+                                </label>
                             </td>
-                            <td class="other-pro-treat">
-                                <obs conceptId="CIEL:164384" allowFutureDates="true" />
+                            <td>
+                                <label>
+                                    <uimessage code="pihcore.hiv.statusAndOrder" />
```
Comment (lnball) [inline configuration/pih/htmlforms/hiv/section-hiv-plan.xml:515]: Same here.  Don't include hiv in the message code.

File: configuration/pih/htmlforms/hiv/section-hiv-plan.xml (hunk 1 above)
Comment (brandones) [reply configuration/pih/htmlforms/hiv/section-hiv-plan.xml:544]: Please don't use `emr.` message codes; those are from the old `openmrs-module-emr` and are defunct and untranslatable. Fortunately all of them seem to have been imported into [coreapps](https://github.com/openmrs/openmrs-module-coreapps/blob/master/api/src/main/resources/messages.properties), so you can just replace `emr.` with `coreapps.` any time you see it. So here I'll use `coreapps.continue`—thanks!

File: configuration/pih/htmlforms/hiv/section-hiv-plan.xml (hunk 2 above)
Comment (brandones) [reply configuration/pih/htmlforms/hiv/section-hiv-plan.xml:540]: I think that the imperative "start" and past participle "started" convey meaningfully different things. The past participle is what is used in the French.

**Review by lnball — COMMENTED** [review]
Just tweaks to the message codes.  In general, for a general label (ie. started, continue, stop, dosage), don't include hiv in the code.  pihcore.[somethingSimple] is enough.  And if it's already in the general OpenMRS, use that.

Other than these small gripes, looks good.

**Pull request merged**